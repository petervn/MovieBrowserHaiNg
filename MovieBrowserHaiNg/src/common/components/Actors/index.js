import React from 'react';
import {FlatList} from 'react-native';

const Actors = ({data, renderItem}) => (
  <FlatList
    data={data}
    horizontal
    showsHorizontalScrollIndicator={false}
    keyExtractor={item => item.id.toString()}
    renderItem={({item}) => renderItem(item)}
  />
);

export default Actors;
