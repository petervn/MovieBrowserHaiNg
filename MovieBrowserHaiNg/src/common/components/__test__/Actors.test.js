import React from 'react';
import {shallow} from 'enzyme';
import Actors from '../Actors';

describe('Actor', () => {
  describe('Rendering', () => {
    const data = [];
    it('should match to snapshot', () => {
      const component = shallow(<Actors data={data} onPress={jest.fn()} />);
      expect(component).toMatchSnapshot();
    });
  });
});
