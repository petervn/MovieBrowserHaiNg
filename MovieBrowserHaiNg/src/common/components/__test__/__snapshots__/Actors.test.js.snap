// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actor Rendering should match to snapshot 1`] = `
<FlatList
  data={Array []}
  disableVirtualization={false}
  horizontal={true}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
